api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1725,17 +1725,27 @@ type AzurePlatformSpec struct {
 
 	// ResourceGroupName is the name of an existing resource group where all cloud resources created by the Hosted
 	// Cluster are to be placed.
+	// Resource group naming requirements can be found here: https://azure.github.io/PSRule.Rules.Azure/en/rules/Azure.ResourceGroup.Name/.
 	// In ARO HCP, this will be the managed resource group where customer cloud resources will be created.
 	//
 	// Example: if your resource group ID is /subscriptions/<subscriptionID>/resourceGroups/<resourceGroupName>, your
 	//          ResourceGroupName is <resourceGroupName>.
 	//
 	// +kubebuilder:default:=default
 	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern=`^[a-zA-Z0-9_()\-\.]{1,89}[a-zA-Z0-9_()\-]$`
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="ResourceGroupName is immutable"
 	// +immutable
 	ResourceGroupName string `json:"resourceGroup"`
 
+	// NetworkResourceGroupName is the name of an existing resource group provided by a client; it is expected to contain
+	// an existing, pre-setup VNET, Subnet and NSG. This field is only used by ARO HCP.
+	//
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="NetworkResourceGroupName is immutable"
+	// +optional
+	// +immutable
+	NetworkResourceGroupName string `json:"networkResourceGroupName,omitempty"`
+
 	// VnetName is the resource name of an existing VNET to use in creating VMs. If this field is included, it should
```

Comment:
For `VnetName` the doc mentions:
```
// VnetName is the resource name of an existing VNET to use in creating VMs. If this field is included, it should
Comment
	// be the resource name matching the VnetID.
	// In ARO HCP, this will be the name of the customer provided VNET.
```

However, the attribute is marked as required. `If this field is included` is incorrect. The field has to be included as it is required, right?

Reply:
This has been updated